crates/bevy_wgpu/src/renderer/wgpu_render_resource_context.rs

```diff
@@ -568,4 +568,21 @@ impl RenderResourceContext for WgpuRenderResourceContext {
             size
         }
     }
+
+    fn get_specialized_shader(&self, shader: &Shader, macros: Option<&[String]>) -> Shader {
+        let macros: Vec<String> = macros
+            .unwrap_or(&[])
+            .iter()
+            .chain((&["WGPU".to_string()]).iter())
```

Comment:
i think id prefer to not insert shader defs for backends (as that encourages writing shaders that are backend-specific). if you need to do that for WebGL backend shaders, could you instead use something like:

```glsl
# ifdef WEBGL
// do webgl things
# endif

# ifndef WEBGL
// do default bevy things
# endif
```

Reply:
done